### PR TITLE
replaced enzyme with react-testing-library

### PR DESCRIPTION
**Pull request by saarkuriel**
### 🔦 Summary
Replaced testing library from `enzyme` to `react-testing-library` in the following generators:

* `client` 
* `client-typescript`
* `server`
* `server-typescript`

### 🗺️ Test Plan
...

### CODE REVIEW

**Comment by netanelgilad** [comment]
@saarkuriel From your experience, what do you think about the setup mentioned in [`react-testing-library` docs](https://testing-library.com/docs/react-testing-library/setup)? 

Maybe we should include it in the generated projects? The cleanup stuff looks useful.

File: packages/create-yoshi-app/templates/client/src/components/App/App.spec.js [hunk 1]
```diff
@@ -1,21 +1,19 @@
-import React from 'react';
-import { mount } from 'enzyme';
+import * as React from 'react';
```
Comment (ranyitz) [inline packages/create-yoshi-app/templates/client/src/components/App/App.spec.js]: Is that intentional?

File: packages/create-yoshi-app/templates/fullstack/src/components/App/App.spec.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-    wrapper = mount(
+    const { queryByTestId } = render(
       <I18nextProvider i18n={i18n}>
         <App />
       </I18nextProvider>,
     );
 
-    expect(wrapper.find('h2')).toHaveLength(1);
+    expect(queryByTestId('app-title')).not.toBeNull();
+    expect(queryByTestId('app-title')).toBeDefined();
```
Comment (ranyitz) [inline packages/create-yoshi-app/templates/fullstack/src/components/App/App.spec.js]: I think that's can be removed

File: packages/create-yoshi-app/templates/client/src/components/App/App.spec.js (hunk 1 above)
Comment (saarkuriel) [reply packages/create-yoshi-app/templates/client/src/components/App/App.spec.js]: nope. fixing

File: packages/create-yoshi-app/templates/fullstack/src/components/App/App.spec.js (hunk 2 above)
Comment (saarkuriel) [reply packages/create-yoshi-app/templates/fullstack/src/components/App/App.spec.js]: correct. removing

**Comment by saarkuriel** [comment]
> @saarkuriel From your experience, what do you think about the setup for `jest` mentioned in [`react-testing-library` docs](https://testing-library.com/docs/react-testing-library/setup)?
> 
> Maybe we should include it in the generated projects? The cleanup stuff looks useful.

I wasn't familiar with this, it looks great! On the one hand it looks like the best place to add it to reduce boilerplate is in `jest-yoshi-preset`, but since `jest-yoshi-preset` is a separate package it will break those who don't have `react-testing-library`. 

One possible solution is to add it in each template we just changed in `jest-yoshi.config`.

WDYT?

**Comment by ranyitz** [comment]
@netanelgilad @saarkuriel I think [that's](https://testing-library.com/docs/react-testing-library/setup#cleanup) a great idea to reduce boilerplate!

I've opened a [separate issue](https://github.com/wix/yoshi/issues/1139) so we could merge this one and continue working on the other.

**Pull request merged**